src/Arcus.WebApi.OpenApi.Extensions/OAuthAuthorizeOperationFilter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Arcus.WebApi.OpenApi.Extensions
+{
+    /// <summary>
+    /// A Swashbuckle operation filter that adds OAuth security definitions to authorized API operations.
+    /// </summary>
+    public class OAuthAuthorizeOperationFilter : IOperationFilter
+    {
+        private readonly IEnumerable<string> _scopes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OAuthAuthorizeOperationFilter"/> class.
+        /// </summary>
+        public OAuthAuthorizeOperationFilter(IEnumerable<string> scopes)
+        {
+            _scopes = scopes;
+        }
+
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            var hasAuthorize = context.MethodInfo.GetCustomAttributes(true).OfType<AuthorizeAttribute>().Any() ||
```

Comment:
Question: now only the built-in authorization attribute is used. Could that be extended with the security attributes that are also part of this project? (shared access key, ...)?

Reply:
We have discussed this and will exclude it in the first iteration and see how it goes, we will probably provide something on top of this one to work with our own ones. We don't want people to force the dependency if they don't need it.